src/test/java/io/confluent/idesidecar/restapi/util/WebClientFactoryTest.java

```diff
@@ -1,14 +1,98 @@
 package io.confluent.idesidecar.restapi.util;
 
 
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+import io.confluent.idesidecar.restapi.application.SidecarInfo;
+import io.quarkiverse.wiremock.devservice.ConnectWireMock;
 import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectSpy;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
 import jakarta.inject.Inject;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 @QuarkusTest
+@ConnectWireMock
 public class WebClientFactoryTest {
   @Inject WebClientFactory webClientFactory;
+  WireMock wireMock;
+  WireMockServer wireMockServer;
+
+  @InjectSpy
+  SidecarInfo sidecarInfo;
+
+  @BeforeEach
+  public void setup() {
+    wireMockServer = new WireMockServer(options().dynamicPort());
+    wireMockServer.start();
+    WireMock.configureFor("localhost", wireMockServer.port());
+  }
+  @AfterEach
+  void resetWireMock() {
+    wireMockServer.stop();
+    wireMock.removeMappings();
+  }
+
+  @Test
+  void getDefaultWebClientOptionsIncludesUserAgent() {
+    var userAgent = webClientFactory.getDefaultWebClientOptions().getUserAgent();
+    Assertions.assertTrue(userAgent.contains("support@confluent.io) sidecar/"));
+  }
+
+
+  @Test
+  public void webClientOptionsShouldContainUserAgentOnCreation() {
+    UrlPattern urlPattern = urlEqualTo("/some-endpoint");
+    // Register the WireMock stub
+    wireMockServer.stubFor(
+        WireMock.get(urlPattern)
+                .willReturn(
+                    aResponse()
+                        .withStatus(200)
+                        .withBody("{\"message\":\"success\"}")
+                )
+    );
+
+    // Create a WebClient instance with the correct port
+    WebClient webClient = WebClient.create(
+        Vertx.vertx(), new WebClientOptions()
+            .setDefaultHost("localhost")
+            .setUserAgent("Confluent-for-VSCode/v%s (https://confluent.io; support@confluent.io) sidecar/v%s (%s/%s)"
+                .formatted(sidecarInfo.vsCode(),sidecarInfo.version(),sidecarInfo.osType(), sidecarInfo.osArch()))
+            .setDefaultPort(wireMockServer.port()));
```

Comment:
```suggestion
    WebClient webClient = WebClient.create(
        Vertx.vertx(), 
        new WebClientOptions()
            .setDefaultHost("localhost")
            .setUserAgent(
                "Confluent-for-VSCode/v%s (https://confluent.io; support@confluent.io) sidecar/v%s (%s/%s)".formatted(
                    sidecarInfo.vsCode()
                    sidecarInfo.version()
                    sidecarInfo.osType(),
                    sidecarInfo.osArch()
                )
            )
            .setDefaultPort(wireMockServer.port())
    );
```
And in fact you should be able to just do this:
```
WebClient webClient = webClientFactory.getWebClient();
webClient
    .getAbs(wireMockServer.baseUrl() + "/some-endpoint")
    .send()
 ...

Reply:
`sidecarInfo.vsCode()` cannot be passed as-is since it returns an `Optional` instance. See my [comment](https://github.com/confluentinc/ide-sidecar/pull/202/files#r1868290337) above. That said, using `webClientFactory.getWebClient()` along with [my other recommendation](https://github.com/confluentinc/ide-sidecar/pull/202/files#r1868334337) to setup the environment should do the trick rather than constructing the user agent here.